hyperspy/signal.py

```diff
@@ -3946,12 +3955,23 @@ def _apply_function_on_data_and_remove_axis(self, function, axes,
         s = out or self._deepcopy_with_new_data(None)
 
         if np.ma.is_masked(self.data):
-            return self._ma_workaround(s, function, axes, ar_axes, out)
+            return self._ma_workaround(s=s, function=function, axes=axes,
+                                       ar_axes=ar_axes, out=out)
         if out:
-            function(self.data, axis=ar_axes, out=out.data)
+            if np_out:
+                function(self.data, axis=ar_axes, out=out.data,)
+            else:
+                data = np.atleast_1d(function(self.data, axis=ar_axes,))
+                if data.shape == out.data.shape:
```

Comment:
This manual check is not necessary, since the bug is only present if `data` is a scalar. Now, since it's an array (because of the `np.atleast_1d`), numpy raises appropriate warning on its own